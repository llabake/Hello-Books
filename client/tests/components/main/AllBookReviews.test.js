import React from 'react';

import {
  AllBookReviews,
  mapDispatchToProps,
  mapStateToProps
} from '../../../components/main/AllBookReviews';

import adminData from '../../mocks/userData';
import userData from '../../mocks/userData';

function setup() {
  const props = {
    dispatch: jest.fn(),
    user: adminData.admin,
    deleteBookReview: jest.fn().mockImplementation(() => Promise.resolve()),
    handleDelete: jest.fn(),
    handleLoadReviewToEdit: jest.fn(),
    loadReviewToEditAction: jest.fn(),
    errors: {},
    reviews: [{
      id: 6,
      content: "updatedAtupdatedAt",
      userId: 6,
      caption: "updatedAtupdatedAt",
      createdAt: "2018-03-19T16:32:57.605Z",
      updatedAt: "2018-03-19T16:32:57.605Z",
      user: {
        id: 2,
        image: "IMAGE",
        username: "mama"
      }
    }]
  };

  const wrapper = shallow(<AllBookReviews {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('<AllBookReview /> component', () => {
  test('should render and not crash', () => {
    const { wrapper } = setup();


    expect(wrapper).toMatchSnapshot();
  })

  test('should handle review deletion', () => {
    const { wrapper, props } = setup();

    props.deleteBookReview.mockReset();
    wrapper.instance().handleDelete(1, 1)
    expect(props.deleteBookReview).toHaveBeenCalled()
  });


  test('should load review to be edited', () => {
    const { wrapper, props } = setup();

    props.loadReviewToEditAction.mockReset();
    wrapper.instance().handleLoadReviewToEdit(1)
    expect(props.loadReviewToEditAction).toHaveBeenCalled()
  });

  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('loadReviewToEditAction');
    expect(mapDispatchToProps(dispatch)).toHaveProperty('deleteBookReview');

    const { deleteBookReview, loadReviewToEditAction } = mapDispatchToProps(dispatch);
    deleteBookReview();
    loadReviewToEditAction();
    expect(dispatch).toHaveBeenCalled();
  });

})