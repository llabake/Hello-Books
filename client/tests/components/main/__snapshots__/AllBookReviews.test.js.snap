// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllBookReview /> component should render and not crash 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col l10 offset-l1 m12 s12"
      key="0"
    >
      <div
        className="row"
      >
        <div
          className="col s4"
        >
          <div
            className="col s4"
          >
            <img
              alt="reviewee image"
              className="circle right"
              height="55"
              src="/IMAGE"
              valign="top"
              width="55"
            />
          </div>
          <div
            className="col s8"
          >
            <span
              style={
                Object {
                  "color": "21282d",
                  "fontSize": ".9375rem",
                  "fontStyle": "normal",
                  "fontWeight": 400,
                  "lineHeight": "1.5",
                }
              }
            >
              mama
            </span>
            <br />
            <span
              style={
                Object {
                  "color": "21282d",
                  "fontSize": ".9375rem",
                  "fontStyle": "normal",
                  "fontWeight": 400,
                  "lineHeight": "1.5",
                }
              }
            >
              3 months ago
            </span>
          </div>
        </div>
        <div
          className="col s8"
        >
          <div>
            <span
              style={
                Object {
                  "fontFamily": "'Lato', sans-serif",
                  "fontSize": "1.0625rem",
                  "fontStyle": "normal",
                  "fontWeight": 700,
                  "lineHeight": "32px",
                  "margin": "0",
                  "maxWidth": "26rem",
                  "wordWrap": "break-word",
                }
              }
            >
              updatedAtupdatedAt
            </span>
            <br />
            <p
              style={
                Object {
                  "color": "21282d",
                  "fontSize": ".9375rem",
                  "fontStyle": "normal",
                  "fontWeight": 400,
                  "lineHeight": "1.5",
                }
              }
            >
              updatedAtupdatedAt
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
